src/youtubeStream.py

```diff
@@ -0,0 +1,123 @@
+import time
+import cv2
+from yt_dlp import YoutubeDL
+
+from typing import ClassVar, Dict, Mapping, Optional, Any, Tuple, List
+from viam.media.video import ViamImage, Optional, CameraMimeType, NamedImage
+from typing_extensions import Self
+from viam.proto.common import ResponseMetadata
+from viam.components.camera import Camera
+from viam.logging import getLogger
+from viam.proto.app.robot import ComponentConfig
+from viam.resource.base import ResourceBase, ResourceName
+from viam.resource.types import Model, ModelFamily
+from viam.utils import struct_to_dict
+from viam.module.types import Reconfigurable
+
+
+LOGGER = getLogger(__name__)
+
+class youtubeStream(Camera, Reconfigurable):
+    
+    """
+    Camera represents any physical hardware that can capture frames.
+    """
+    video_url: str
+    default_url = "https://www.youtube.com/watch?v=MusSS4R9SPw"
+
+    MODEL: ClassVar[Model] = Model(ModelFamily("julie", "camera"), "youtubeStream")
+    REQUIRED_ATTRIBUTES = ["video_url"]
+    
+
+    # Constructor
+    @classmethod
+    def new(cls, config: ComponentConfig, dependencies: Mapping[ResourceName, ResourceBase]) -> Self:
+        my_class = cls(config.name)
+        my_class.reconfigure(config, dependencies)
+        return my_class
+    
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.last_call_time = time.time()
+        self.current_frame = 0
+        self.video_cap = None
```

Comment:
Do we ever construct with an empty video capture, if so we should defensively check for `None` later on in our code.

Reply:
nit: recommend type hinting all instance and class attributes e.g.

```
current_frame: int
video_camp: Optional[<whatever-type-it-is>]
```
etc.